### PR TITLE
Updates code base to 3.6+ compatibility

### PR DESCRIPTION
**Pull request by Shreshtth**

### CODE REVIEW

File: src/config/config.py [hunk 1]
```diff
@@ -248,9 +248,9 @@ def parse_configs():
     configs.use_best_checkpoint = True
 
     if configs.use_best_checkpoint:
-        configs.saved_weight_name = os.path.join(configs.checkpoints_dir, '{}_best.pth'.format(configs.saved_fn))
+        configs.saved_weight_name = os.path.join(configs.checkpoints_dir, f'{configs.saved_fn}_best.pth')
     else:
-        configs.saved_weight_name = os.path.join(configs.checkpoints_dir, '{}.pth'.format(configs.saved_fn))
+        configs.saved_weight_name = os.path.join(configs.checkpoints_dir, f'{configs.saved_fn}.pth')
```
Comment (onlinejudge95) [inline src/config/config.py]: We need to use `pathlib` APIs for managing OS paths here

File: src/data_process/ttnet_data_utils.py [hunk 2]
```diff
@@ -108,23 +108,23 @@ def get_events_infor(game_list, configs, dataset_type):
                                                                  smooth_idx + num_frames_from_event + 1)]
                 img_path_list = []
                 for sub_smooth_idx in sub_smooth_frame_indices:
-                    img_path = os.path.join(images_dir, game_name, 'img_{:06d}.jpg'.format(sub_smooth_idx))
+                    img_path = os.path.join(images_dir, game_name, f'img_{sub_smooth_idx:06d}.jpg')
```
Comment (onlinejudge95) [inline src/data_process/ttnet_data_utils.py]: Same as above

File: src/data_process/ttnet_data_utils.py [hunk 3]
```diff
@@ -108,23 +108,23 @@ def get_events_infor(game_list, configs, dataset_type):
...
-                ball_position_xy = ball_annos['{}'.format(last_f_idx)]
+                ball_position_xy = ball_annos[f'{last_f_idx}']
                 ball_position_xy = np.array([ball_position_xy['x'], ball_position_xy['y']], dtype=np.int)
                 # Ignore the event without ball information
                 if (ball_position_xy[0] < 0) or (ball_position_xy[1] < 0):
                     continue
 
                 # Get segmentation path for the last frame in the sequence
-                seg_path = os.path.join(annos_dir, game_name, 'segmentation_masks', '{}.png'.format(last_f_idx))
+                seg_path = os.path.join(annos_dir, game_name, 'segmentation_masks', f'{last_f_idx}.png')
```
Comment (onlinejudge95) [inline src/data_process/ttnet_data_utils.py]: Same as above

File: src/data_process/ttnet_dataset.py [hunk 4]
```diff
@@ -136,31 +136,27 @@ def __getitem__(self, index):
...
-        axes[i].set_title('image {}'.format(i))
+        axes[i].set_title(f'image {i}')
     fig.suptitle(
-        'Event: is bounce {}, is net: {}, ball_position_xy: (x= {}, y= {})'.format(target_event[0], target_event[1],
-                                                                                   org_ball_pos_xy[0],
-                                                                                   org_ball_pos_xy[1]),
+        f'Event: is bounce {target_event[0]}, is net: {target_event[1]}, ball_position_xy: (x= {org_ball_pos_xy[0]}, y= {org_ball_pos_xy[1]})',
         fontsize=16)
-    plt.savefig(os.path.join(out_images_dir, 'org_all_imgs_{}.jpg'.format(example_index)))
+    plt.savefig(os.path.join(out_images_dir, f'org_all_imgs_{example_index}.jpg'))
```
Comment (onlinejudge95) [inline src/data_process/ttnet_dataset.py]: Same as above

File: src/data_process/ttnet_dataset.py [hunk 5]
```diff
@@ -136,31 +136,27 @@ def __getitem__(self, index):
...
     target_seg = target_seg.transpose(1, 2, 0)
-    print('target_seg shape: {}'.format(target_seg.shape))
+    print(f'target_seg shape: {target_seg.shape}')
 
-    plt.imsave(os.path.join(out_images_dir, 'augment_seg_img_{}.jpg'.format(example_index)), target_seg)
+    plt.imsave(os.path.join(out_images_dir, f'augment_seg_img_{example_index}.jpg'), target_seg)
```
Comment (onlinejudge95) [inline src/data_process/ttnet_dataset.py]: Same as above

File: src/data_process/ttnet_data_utils.py [hunk 6]
```diff
@@ -108,23 +109,23 @@ def get_events_infor(game_list, configs, dataset_type):
...
+                ball_position_xy = ball_annos[f'{last_f_idx}']
                 ball_position_xy = np.array([ball_position_xy['x'], ball_position_xy['y']], dtype=np.int)
                 # Ignore the event without ball information
                 if (ball_position_xy[0] < 0) or (ball_position_xy[1] < 0):
                     continue
 
                 # Get segmentation path for the last frame in the sequence
-                seg_path = os.path.join(annos_dir, game_name, 'segmentation_masks', '{}.png'.format(last_f_idx))
+                seg_path = Path(annos_dir) / game_name / 'segmentation_masks' / f'{last_f_idx}.png'
                 if not os.path.isfile(seg_path):
```
Comment (onlinejudge95) [inline src/data_process/ttnet_data_utils.py]: This needs to be migrated as well.

File: src/data_process/ttnet_video_loader.py [hunk 7]
```diff
@@ -20,7 +21,7 @@ class TTNet_Video_Loader:
     """The loader for demo with a video input"""
 
     def __init__(self, video_path, input_size=(320, 128), num_frames_sequence=9):
-        assert os.path.isfile(video_path), "No video at {}".format(video_path)
+        assert os.path.isfile(video_path), f"No video at {video_path}"
```
Comment (onlinejudge95) [inline src/data_process/ttnet_video_loader.py]: This also needs to be migrated

File: src/data_process/ttnet_video_loader.py [hunk 8]
```diff
@@ -75,32 +76,32 @@ def __len__(self):
...
     configs = parse_configs()
 
-    video_path = os.path.join(configs.dataset_dir, 'test', 'videos', 'test_1.mp4')
+    video_path = Path(configs.dataset_dir) / 'test' / 'videos' / 'test_1.mp4'
     video_loader = TTNet_Video_Loader(video_path, input_size=(320, 128),
                                       num_frames_sequence=configs.num_frames_sequence)
-    out_images_dir = os.path.join(configs.results_dir, 'debug', 'ttnet_video_loader')
+    out_images_dir = Path(configs.results_dir) / 'debug'/ 'ttnet_video_loader'
     if not os.path.isdir(out_images_dir):
         os.makedirs(out_images_dir)
```
Comment (onlinejudge95) [inline src/data_process/ttnet_video_loader.py]: This needs to be migrated as well

File: src/demo.py [hunk 9]
```diff
@@ -28,14 +29,14 @@
...
 def demo(configs):
     video_loader = TTNet_Video_Loader(configs.video_path, configs.input_size, configs.num_frames_sequence)
-    result_filename = os.path.join(configs.save_demo_dir, 'results.txt')
+    result_filename = Path(configs.save_demo_dir) / 'results.txt'
     frame_rate = video_loader.video_fps
     if configs.save_demo_output:
-        configs.frame_dir = os.path.join(configs.save_demo_dir, 'frame')
+        configs.frame_dir = Path(configs.save_demo_dir) / 'frame'
         if not os.path.isdir(configs.frame_dir):
             os.makedirs(configs.frame_dir)
```
Comment (onlinejudge95) [inline src/demo.py]: This needs to be migrated as well

File: src/models/model_utils.py [hunk 10]
```diff
@@ -81,12 +81,12 @@ def load_weights_local_stage(pretrained_dict):
 
 def load_pretrained_model(model, pretrained_path, gpu_idx, overwrite_global_2_local):
     """Load weights from the pretrained model"""
-    assert os.path.isfile(pretrained_path), "=> no checkpoint found at '{}'".format(pretrained_path)
+    assert os.path.isfile(pretrained_path), f"=> no checkpoint found at '{pretrained_path}'"
```
Comment (onlinejudge95) [inline src/models/model_utils.py]: This needs to be migrated as well

File: src/models/model_utils.py [hunk 11]
```diff
@@ -116,15 +116,15 @@ def load_pretrained_model(model, pretrained_path, gpu_idx, overwrite_global_2_lo
 
 def resume_model(resume_path, arch, gpu_idx):
     """Resume training model from the previous trained checkpoint"""
-    assert os.path.isfile(resume_path), "=> no checkpoint found at '{}'".format(resume_path)
+    assert os.path.isfile(resume_path), f"=> no checkpoint found at '{resume_path}'"
```
Comment (onlinejudge95) [inline src/models/model_utils.py]: This needs to be migrated as well

File: src/utils/init_paths.py [hunk 12]
```diff
@@ -13,5 +14,5 @@ def add_path(self, path):
     def add_lib(self, libs_list):
         # Add lib to PYTHONPATH
         for lib in libs_list:
-            lib_path = os.path.join(self.this_dir, lib)
+            lib_path = Path(self.this_dir) / lib
             self.add_path(lib_path)
```
Comment (onlinejudge95) [inline src/utils/init_paths.py]: This can be merged in a single line

File: src/utils/logger.py [hunk 13]
```diff
@@ -14,8 +14,8 @@ class Logger():
         """
 
     def __init__(self, logs_dir, saved_fn):
-        logger_fn = 'logger_{}.txt'.format(saved_fn)
-        logger_path = os.path.join(logs_dir, logger_fn)
+        logger_fn = f'logger_{saved_fn}.txt'
+        logger_path = Path(logs_dir) / logger_fn
```
Comment (onlinejudge95) [inline src/utils/logger.py]: These can be compressed in a single line

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

File: src/data_process/ttnet_data_utils.py [hunk 14]
```diff
@@ -108,23 +108,23 @@ def get_events_infor(game_list, configs, dataset_type):
...
                 # Ignore the event without ball information
                 if (ball_position_xy[0] < 0) or (ball_position_xy[1] < 0):
                     continue
 
                 # Get segmentation path for the last frame in the sequence
-                seg_path = os.path.join(annos_dir, game_name, 'segmentation_masks', '{}.png'.format(last_f_idx))
-                if not os.path.isfile(seg_path):
-                    print("smooth_idx: {} - The segmentation path {} is invalid".format(smooth_idx, seg_path))
+                seg_path = Path(annos_dir) / game_name / 'segmentation_masks' / f'{last_f_idx}.png'
+                if not Path(seg_path).is_file():
```
Comment (onlinejudge95) [inline src/data_process/ttnet_data_utils.py]: `seg_path` is already a `Path` object.

I see a lot of occurrences for this, please fix all of them

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

**Review by Shreshtth — COMMENTED** [review]
@onlinejudge95 Please review these changes once.

**Review by onlinejudge95 — APPROVED** [review]

**Pull request merged**